sift/src/main/java/siftscience/android/Uploader.java

```diff
@@ -1,357 +1,213 @@
+                connection.setRequestProperty("Content-Type", "application/json");
+                connection.setFixedLengthStreamingMode(this.requestBody.length);
+                connection.setDoOutput(true);
+                connection.setDoInput(true);
+
+                connection.connect();
+
+                try {
+                    OutputStream out = connection.getOutputStream();
+
                     try {
-                        state.request = makeRequest();
-                    } catch (IOException e) {
-                        Log.e(TAG, "Encountered IOException in makeRequest", e);
+                        out.write(this.requestBody);
+                    } finally {
+                        out.close();
                     }
-                }
-                if (state.request == null) {
-                    // Nothing to upload at the moment; check again a minute later
-                    safeSchedule(checkState, 60, TimeUnit.SECONDS);
-                    return;
-                }
 
-                long now = Time.now();
-                if (state.nextUploadTime > now) {
-                    // We've woken up too early; go back to sleep
-                    long delay = state.nextUploadTime - now;
-                    safeSchedule(checkState, delay, TimeUnit.MILLISECONDS);
-                    return;
-                }
+                    int code = connection.getResponseCode();
+                    String body = null;
+                    InputStream in;
 
-                try {
-                    Log.d(TAG, "Send HTTP request");
-                    // try-with needs API level 19+ :(
-                    Response response = client.newCall(state.request).execute();
-                    int code = response.code();
-                    String body = response.body() == null ? null : response.body().string();
-                    response.close();
+                    if (code >= 400) {
+                        in = connection.getErrorStream();
+                    } else {
+                        in = connection.getInputStream();
+                    }
 
-                    if (code == 200) {
-                        // Success, reset the state and move on to the next batch
-                        state.reset();
-                        batches.pop();
-                        safeSubmit(checkState);
-
-                        // This is for testing
-                        if (onRequestCompletion != null) {
-                            onRequestCompletion.release();
+                    if (in != null) {
+                        try {
+                            body = readInputStreamAsString(in, MAX_BYTES);
+                        } finally {
+                            in.close();
                         }
+                    }
 
+                    if (code == 200) {
+                        Log.d(TAG,"HTTP 200");
                         return;
```

Comment:
Nit: Intellij said this return is redundant

Reply:
removed